src/NexusMods.CrossPlatform/ProtocolRegistration/ProtocolRegistrationLinux.cs

```diff
@@ -125,15 +131,101 @@ private async Task UpdateMIMECacheDatabase(AbsolutePath applicationsDirectory, C
+            .Replace("$", @"\$"); // dollar sign ("$")                             $ -> \$
+
+        // First apply the base rules from `string` and `localstring`.
+        var escapedFinalPath = EscapeDesktopFilePath(escapedPath);
+
+        // Note(sewer): Quoting the spec
+        // 
+        // > Note that the general escape rule for values of type string states that the backslash
+        //   character can be escaped as ("\\") as well and that this escape rule is applied before
+        //   the quoting rule. As such, to unambiguously represent a literal backslash character in
+        //   a quoted argument in a desktop entry file requires the use of four successive backslash
+        //   characters ("\\\\").
+        // 
+        // So escaping `\` twice, leading to 4 backslashes as a result of applying both functions is by design,
+        // even if it may 'feel weird'.
+        //
+        // > Likewise, a literal dollar sign in a quoted argument in a desktop
+        //   entry file is unambiguously represented with ("\\$"). 
+        //
+        // So we go from `$` to `\$`.
+        // And then from `\$` to `\\$`.
+        // 
+        // As per the example in the spec, this is not a bug, this is intended behaviour, even if weird.
+
+        // Note(sewer):
+        //
+        // The docs say:
+        // > Arguments may be quoted in whole. If an argument contains a reserved character the argument must be quoted.
+        //
+        // > Reserved characters are space (" "), tab, newline, double quote, single quote ("'"), backslash character ("\"),
+        // > greater-than sign (">"), less-than sign ("<"), tilde ("~"), vertical bar ("|"), ampersand ("&"), semicolon (";"),
+        // > dollar sign ("$"), asterisk ("*"), question mark ("?"), hash mark ("#"), parenthesis ("(") and (")") and backtick character ("`").
+        //
+        // In this case, we always quote. Quoting is optional if there are no 'reserved' characters,
+        // and mandatory if there are. By always quoting, risk is reduced.
+        return $"\"{escapedFinalPath}\""; // Enclose the entire path in double quotes
+    }
+
+    // Note(sewer)
+    // For other fields 'string' and 'localestring':
+    //
+    // https://specifications.freedesktop.org/desktop-entry-spec/1.0/value-types.html
+    // The escape sequences \s, \n, \t, \r, and \\ are supported for values of type string and localestring,
+    // meaning ASCII space, newline, tab, carriage return, and backslash, respectively. 
+    //
+    // Note that 'Exec' is a string https://specifications.freedesktop.org/desktop-entry-spec/1.0/recognized-keys.html
+    private string EscapeDesktopFilePath(string path)
+    {
+        return path
+            .Replace(@"\", @"\\") // Escape backslashes first to avoid double escaping
+            .Replace(" ", @"\s")
+            .Replace("\n", @"\n")
+            .Replace("\t", @"\t")
+            .Replace("\r", @"\r");
+    }
```

Comment:
Could you make these `internal static` and add tests?

Reply:
I added tests for every single escape character mentioned in the relevant spec sections.
And notes regarding some others like `%` and `;` which don't apply in our case, including why they don't apply; all within spec.